packages/tools/lib/create-new-component/index.js

```diff
@@ -108,47 +35,102 @@ const getLibraryName = packageName => {
 	return packageName.substr("webcomponents-".length);
 };
 
-const camelToKebabCase = string => string.replace(/([a-z])([A-Z])/g, "$1-$2").toLowerCase();
-
-const packageName = getPackageName();
-const library = getLibraryName(packageName);
-
-const consoleArguments = process.argv.slice(2);
-const componentName = consoleArguments[0];
-
-if (!componentName){
-	console.error("Please enter component name.");
-	return;
+// String manipulation
+const capitalizeFirstLetter = string => string.charAt(0).toUpperCase() + string.slice(1);
+
+// Validation of user input
+const isNameValid = name => typeof name === "string" && name.match(/^[a-zA-Z][a-zA-Z0-9_-]*$/);
+const isTagNameValid = tagName => tagName.match(/^[a-z][a-z0-9]+-[a-z0-9]*(-[a-z0-9]+)*$/);
+
+const generateFiles = (componentName, tagName, library, packageName, isTypeScript) => {
+	componentName = capitalizeFirstLetter(componentName);
+	const filePaths = {
+		"main": isTypeScript 
+			? `./src/${componentName}.ts` 
+			: `./src/${componentName}.js`,
+		"css": `./src/themes/${componentName}.css`,
+		"template": `./src/${componentName}.hbs`,
+	};
+
+	const FileContentTemplate = isTypeScript 
+		? tsFileContentTemplate(componentName, tagName, library, packageName) 
+		: jsFileContentTemplate(componentName, tagName, library, packageName);
+
+	fs.writeFileSync(filePaths.main, FileContentTemplate, { flag: "wx+" });
+	fs.writeFileSync(filePaths.css, "", { flag: "wx+" });
+	fs.writeFileSync(filePaths.template, "<div>Hello World</div>", { flag: "wx+" });
+
+	console.log(`Successfully generated ${filePaths.main}`);
+	console.log(`Successfully generated ${filePaths.css}`);
+	console.log(`Successfully generated ${filePaths.template}`);
+
+	// Change the color of the output
+		console.warn('\x1b[33m%s\x1b[0m', `
+	Make sure to import the component in your bundle by using:
+	import ${componentName} from "./dist/${componentName}.js";`);
 }
 
-const tagName = `ui5-${camelToKebabCase(componentName)}`;
-
-const filePaths = {
-	"js": `./src/${componentName}.js`,
-	"css": `./src/themes/${componentName}.css`,
-	"hbs": `./src/${componentName}.hbs`,
-};
-const sJsFileContentTemplate = jsFileContentTemplate(componentName);
-
-fs.writeFileSync(filePaths.js, sJsFileContentTemplate, { flag: "wx+" });
-fs.writeFileSync(filePaths.css, "", { flag: "wx+" });
-fs.writeFileSync(filePaths.hbs, "<div>Hello World</div>", { flag: "wx+" });
+// Main function
+const createWebComponent = async () => {
+	const consoleArguments = process.argv.slice(2);
+	let componentName = consoleArguments[0];
+	let tagName = consoleArguments[1];
+	let language = consoleArguments[2];
+	let isTypeScript;
+
```

Comment:
It would be easier to throw errors here.
```
if (componentName && !isValid(componentName)) {.... throw error here ...}
```
same for tag name
same for language

Then, you can start calling the wizards.